teamcity: failed tests on master: acceptance/TestClusterRecovery

The following tests appear to have failed:

[#792103](https://teamcity.cockroachdb.com/viewLog.html?buildId=792103):


```
--- FAIL: acceptance/TestClusterRecovery (36.610s)
test_log_scope.go:81: test logs captured to: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/logTestClusterRecovery561334910
test_log_scope.go:62: use -show-logs to present logs inline
--- FAIL: acceptance/TestClusterRecovery: TestClusterRecovery/runMode=local (36.600s)
zchaos_test.go:152: pq: no inbound stream connection
------- Stdout: -------
CockroachDB node starting at 2018-07-23 05:57:00.171706703 +0000 UTC (took 0.4s)
build:               CCL v2.1.0-alpha.20180702-734-g54a811d @ 2018/07/23 05:45:13 (go1.10.3)
admin:               http://127.0.0.1:26258
sql:                 postgresql://root@127.0.0.1:26257?sslmode=disable
logs:                /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestClusterRecovery/runMode=local/1
temp dir:            /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster589129669/1/cockroach-temp989178689
external I/O path:   /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster589129669/1/extern
store[0]:            path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster589129669/1
status:              initialized new cluster
clusterID:           91c48483-7eac-4918-84e1-8a069a3f3930
nodeID:              1
CockroachDB node starting at 2018-07-23 05:57:00.905598933 +0000 UTC (took 1.2s)
build:               CCL v2.1.0-alpha.20180702-734-g54a811d @ 2018/07/23 05:45:13 (go1.10.3)
admin:               http://127.0.0.1:26264
sql:                 postgresql://root@127.0.0.1:26263?sslmode=disable
logs:                /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestClusterRecovery/runMode=local/4
temp dir:            /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster589129669/4/cockroach-temp005658158
external I/O path:   /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster589129669/4/extern
store[0]:            path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster589129669/4
status:              initialized new node, joined pre-existing cluster
clusterID:           91c48483-7eac-4918-84e1-8a069a3f3930
nodeID:              2
CockroachDB node starting at 2018-07-23 05:57:00.930210127 +0000 UTC (took 1.2s)
build:               CCL v2.1.0-alpha.20180702-734-g54a811d @ 2018/07/23 05:45:13 (go1.10.3)
admin:               http://127.0.0.1:26260
sql:                 postgresql://root@127.0.0.1:26259?sslmode=disable
logs:                /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestClusterRecovery/runMode=local/2
temp dir:            /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster589129669/2/cockroach-temp197772196
external I/O path:   /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster589129669/2/extern
store[0]:            path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster589129669/2
status:              initialized new node, joined pre-existing cluster
clusterID:           91c48483-7eac-4918-84e1-8a069a3f3930
nodeID:              3
CockroachDB node starting at 2018-07-23 05:57:00.938670846 +0000 UTC (took 1.2s)
build:               CCL v2.1.0-alpha.20180702-734-g54a811d @ 2018/07/23 05:45:13 (go1.10.3)
admin:               http://127.0.0.1:26262
sql:                 postgresql://root@127.0.0.1:26261?sslmode=disable
logs:                /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestClusterRecovery/runMode=local/3
temp dir:            /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster589129669/3/cockroach-temp486351459
external I/O path:   /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster589129669/3/extern
store[0]:            path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster589129669/3
status:              initialized new node, joined pre-existing cluster
clusterID:           91c48483-7eac-4918-84e1-8a069a3f3930
nodeID:              4
                            [1]   4
                 liveness-  [2]   1
    liveness-    liveness.  [3]   1
    liveness.   tsd         [4]   4
   tsd          tse         [5]   4
   tse         �            [6]   4
  �            �            [7]   3
  �            �            [8]   4
  �            �            [9]   4
  �            �            [10]  4
  �            �            [11]  4
  �            �            [12]  4
  �            �            [13]  4
  �            �            [14]  4
  �            �            [15]  4
  �            �            [16]  4
  �            �            [17]  4
  �            �            [18]  3
  �            �            [19]  3
  �            �            [20]  4
  �            �            [21]  3
  �            ��             [22]  3
                            [1]   3
                 liveness-  [2]   1
    liveness-    liveness.  [3]   3
    liveness.   tsd         [4]   4
   tsd          tse         [5]   3
   tse         �            [6]   4
  �            �            [7]   3
  �            �            [8]   4
  �            �            [9]   3
  �            �            [10]  3
  �            �            [11]  4
  �            �            [12]  3
  �            �            [13]  4
  �            �            [14]  3
  �            �            [15]  3
  �            �            [16]  4
  �            �            [17]  4
  �            �            [18]  3
  �            �            [19]  3
  �            �            [20]  4
  �            �            [21]  3
  �            ��             [22]  3
                            [1]   3
                 liveness-  [2]   1
    liveness-    liveness.  [3]   3
    liveness.   tsd         [4]   3
   tsd          tse         [5]   3
   tse         �            [6]   3
  �            �            [7]   3
  �            �            [8]   3
  �            �            [9]   3
  �            �            [10]  3
  �            �            [11]  3
  �            �            [12]  3
  �            �            [13]  3
  �            �            [14]  3
  �            �            [15]  3
  �            �            [16]  4
  �            �            [17]  3
  �            �            [18]  3
  �            �            [19]  3
  �            �            [20]  4
  �            �            [21]  3
  �            ��             [22]  3
                            [1]   3
                 liveness-  [2]   3
    liveness-    liveness.  [3]   3
    liveness.   tsd         [4]   3
   tsd          tse         [5]   3
   tse         �            [6]   3
  �            �            [7]   3
  �            �            [8]   3
  �            �            [9]   3
  �            �            [10]  3
  �            �            [11]  3
  �            �            [12]  3
  �            �            [13]  3
  �            �            [14]  3
  �            �            [15]  3
  �            �            [16]  3
  �            �            [17]  3
  �            �            [18]  3
  �            �            [19]  3
  �            �            [20]  3
  �            �            [21]  3
  �            ��             [22]  3
CockroachDB node starting at 2018-07-23 05:57:21.880271889 +0000 UTC (took 0.6s)
build:               CCL v2.1.0-alpha.20180702-734-g54a811d @ 2018/07/23 05:45:13 (go1.10.3)
admin:               http://127.0.0.1:26258
sql:                 postgresql://root@127.0.0.1:26257?sslmode=disable
logs:                /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestClusterRecovery/runMode=local/1
temp dir:            /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster589129669/1/cockroach-temp805717628
external I/O path:   /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster589129669/1/extern
store[0]:            path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster589129669/1
status:              restarted pre-existing node
clusterID:           91c48483-7eac-4918-84e1-8a069a3f3930
nodeID:              1
E180723 05:57:23.130484 6439 acceptance/zchaos_test.go:272  round 1: failed to do consistency check against node 0: Consistency checking is unimplmented and should be re-implemented using SQL
CockroachDB node starting at 2018-07-23 05:57:23.862115933 +0000 UTC (took 0.7s)
build:               CCL v2.1.0-alpha.20180702-734-g54a811d @ 2018/07/23 05:45:13 (go1.10.3)
admin:               http://127.0.0.1:26260
sql:                 postgresql://root@127.0.0.1:26259?sslmode=disable
logs:                /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestClusterRecovery/runMode=local/2
temp dir:            /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster589129669/2/cockroach-temp431361158
external I/O path:   /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster589129669/2/extern
store[0]:            path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster589129669/2
status:              restarted pre-existing node
clusterID:           91c48483-7eac-4918-84e1-8a069a3f3930
nodeID:              3
CockroachDB node starting at 2018-07-23 05:57:24.906061858 +0000 UTC (took 0.7s)
build:               CCL v2.1.0-alpha.20180702-734-g54a811d @ 2018/07/23 05:45:13 (go1.10.3)
admin:               http://127.0.0.1:26262
sql:                 postgresql://root@127.0.0.1:26261?sslmode=disable
logs:                /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestClusterRecovery/runMode=local/3
temp dir:            /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster589129669/3/cockroach-temp973375792
external I/O path:   /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster589129669/3/extern
store[0]:            path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster589129669/3
status:              restarted pre-existing node
clusterID:           91c48483-7eac-4918-84e1-8a069a3f3930
nodeID:              4
CockroachDB node starting at 2018-07-23 05:57:25.923620497 +0000 UTC (took 0.7s)
build:               CCL v2.1.0-alpha.20180702-734-g54a811d @ 2018/07/23 05:45:13 (go1.10.3)
admin:               http://127.0.0.1:26258
sql:                 postgresql://root@127.0.0.1:26257?sslmode=disable
logs:                /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestClusterRecovery/runMode=local/1
temp dir:            /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster589129669/1/cockroach-temp584756995
external I/O path:   /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster589129669/1/extern
store[0]:            path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster589129669/1
status:              restarted pre-existing node
clusterID:           91c48483-7eac-4918-84e1-8a069a3f3930
nodeID:              1
test logs left over in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/logTestClusterRecovery561334910
--- FAIL: acceptance/TestClusterRecovery (36.610s)
test_log_scope.go:81: test logs captured to: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/logTestClusterRecovery561334910
test_log_scope.go:62: use -show-logs to present logs inline
--- FAIL: acceptance/TestClusterRecovery: TestClusterRecovery/runMode=local (36.600s)
zchaos_test.go:152: pq: no inbound stream connection
------- Stdout: -------
CockroachDB node starting at 2018-07-23 05:57:00.171706703 +0000 UTC (took 0.4s)
build:               CCL v2.1.0-alpha.20180702-734-g54a811d @ 2018/07/23 05:45:13 (go1.10.3)
admin:               http://127.0.0.1:26258
sql:                 postgresql://root@127.0.0.1:26257?sslmode=disable
logs:                /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestClusterRecovery/runMode=local/1
temp dir:            /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster589129669/1/cockroach-temp989178689
external I/O path:   /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster589129669/1/extern
store[0]:            path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster589129669/1
status:              initialized new cluster
clusterID:           91c48483-7eac-4918-84e1-8a069a3f3930
nodeID:              1
CockroachDB node starting at 2018-07-23 05:57:00.905598933 +0000 UTC (took 1.2s)
build:               CCL v2.1.0-alpha.20180702-734-g54a811d @ 2018/07/23 05:45:13 (go1.10.3)
admin:               http://127.0.0.1:26264
sql:                 postgresql://root@127.0.0.1:26263?sslmode=disable
logs:                /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestClusterRecovery/runMode=local/4
temp dir:            /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster589129669/4/cockroach-temp005658158
external I/O path:   /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster589129669/4/extern
store[0]:            path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster589129669/4
status:              initialized new node, joined pre-existing cluster
clusterID:           91c48483-7eac-4918-84e1-8a069a3f3930
nodeID:              2
CockroachDB node starting at 2018-07-23 05:57:00.930210127 +0000 UTC (took 1.2s)
build:               CCL v2.1.0-alpha.20180702-734-g54a811d @ 2018/07/23 05:45:13 (go1.10.3)
admin:               http://127.0.0.1:26260
sql:                 postgresql://root@127.0.0.1:26259?sslmode=disable
logs:                /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestClusterRecovery/runMode=local/2
temp dir:            /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster589129669/2/cockroach-temp197772196
external I/O path:   /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster589129669/2/extern
store[0]:            path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster589129669/2
status:              initialized new node, joined pre-existing cluster
clusterID:           91c48483-7eac-4918-84e1-8a069a3f3930
nodeID:              3
CockroachDB node starting at 2018-07-23 05:57:00.938670846 +0000 UTC (took 1.2s)
build:               CCL v2.1.0-alpha.20180702-734-g54a811d @ 2018/07/23 05:45:13 (go1.10.3)
admin:               http://127.0.0.1:26262
sql:                 postgresql://root@127.0.0.1:26261?sslmode=disable
logs:                /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestClusterRecovery/runMode=local/3
temp dir:            /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster589129669/3/cockroach-temp486351459
external I/O path:   /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster589129669/3/extern
store[0]:            path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster589129669/3
status:              initialized new node, joined pre-existing cluster
clusterID:           91c48483-7eac-4918-84e1-8a069a3f3930
nodeID:              4
                            [1]   4
                 liveness-  [2]   1
    liveness-    liveness.  [3]   1
    liveness.   tsd         [4]   4
   tsd          tse         [5]   4
   tse         �            [6]   4
  �            �            [7]   3
  �            �            [8]   4
  �            �            [9]   4
  �            �            [10]  4
  �            �            [11]  4
  �            �            [12]  4
  �            �            [13]  4
  �            �            [14]  4
  �            �            [15]  4
  �            �            [16]  4
  �            �            [17]  4
  �            �            [18]  3
  �            �            [19]  3
  �            �            [20]  4
  �            �            [21]  3
  �            ��             [22]  3
                            [1]   3
                 liveness-  [2]   1
    liveness-    liveness.  [3]   3
    liveness.   tsd         [4]   4
   tsd          tse         [5]   3
   tse         �            [6]   4
  �            �            [7]   3
  �            �            [8]   4
  �            �            [9]   3
  �            �            [10]  3
  �            �            [11]  4
  �            �            [12]  3
  �            �            [13]  4
  �            �            [14]  3
  �            �            [15]  3
  �            �            [16]  4
  �            �            [17]  4
  �            �            [18]  3
  �            �            [19]  3
  �            �            [20]  4
  �            �            [21]  3
  �            ��             [22]  3
                            [1]   3
                 liveness-  [2]   1
    liveness-    liveness.  [3]   3
    liveness.   tsd         [4]   3
   tsd          tse         [5]   3
   tse         �            [6]   3
  �            �            [7]   3
  �            �            [8]   3
  �            �            [9]   3
  �            �            [10]  3
  �            �            [11]  3
  �            �            [12]  3
  �            �            [13]  3
  �            �            [14]  3
  �            �            [15]  3
  �            �            [16]  4
  �            �            [17]  3
  �            �            [18]  3
  �            �            [19]  3
  �            �            [20]  4
  �            �            [21]  3
  �            ��             [22]  3
                            [1]   3
                 liveness-  [2]   3
    liveness-    liveness.  [3]   3
    liveness.   tsd         [4]   3
   tsd          tse         [5]   3
   tse         �            [6]   3
  �            �            [7]   3
  �            �            [8]   3
  �            �            [9]   3
  �            �            [10]  3
  �            �            [11]  3
  �            �            [12]  3
  �            �            [13]  3
  �            �            [14]  3
  �            �            [15]  3
  �            �            [16]  3
  �            �            [17]  3
  �            �            [18]  3
  �            �            [19]  3
  �            �            [20]  3
  �            �            [21]  3
  �            ��             [22]  3
CockroachDB node starting at 2018-07-23 05:57:21.880271889 +0000 UTC (took 0.6s)
build:               CCL v2.1.0-alpha.20180702-734-g54a811d @ 2018/07/23 05:45:13 (go1.10.3)
admin:               http://127.0.0.1:26258
sql:                 postgresql://root@127.0.0.1:26257?sslmode=disable
logs:                /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestClusterRecovery/runMode=local/1
temp dir:            /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster589129669/1/cockroach-temp805717628
external I/O path:   /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster589129669/1/extern
store[0]:            path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster589129669/1
status:              restarted pre-existing node
clusterID:           91c48483-7eac-4918-84e1-8a069a3f3930
nodeID:              1
E180723 05:57:23.130484 6439 acceptance/zchaos_test.go:272  round 1: failed to do consistency check against node 0: Consistency checking is unimplmented and should be re-implemented using SQL
CockroachDB node starting at 2018-07-23 05:57:23.862115933 +0000 UTC (took 0.7s)
build:               CCL v2.1.0-alpha.20180702-734-g54a811d @ 2018/07/23 05:45:13 (go1.10.3)
admin:               http://127.0.0.1:26260
sql:                 postgresql://root@127.0.0.1:26259?sslmode=disable
logs:                /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestClusterRecovery/runMode=local/2
temp dir:            /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster589129669/2/cockroach-temp431361158
external I/O path:   /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster589129669/2/extern
store[0]:            path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster589129669/2
status:              restarted pre-existing node
clusterID:           91c48483-7eac-4918-84e1-8a069a3f3930
nodeID:              3
CockroachDB node starting at 2018-07-23 05:57:24.906061858 +0000 UTC (took 0.7s)
build:               CCL v2.1.0-alpha.20180702-734-g54a811d @ 2018/07/23 05:45:13 (go1.10.3)
admin:               http://127.0.0.1:26262
sql:                 postgresql://root@127.0.0.1:26261?sslmode=disable
logs:                /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestClusterRecovery/runMode=local/3
temp dir:            /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster589129669/3/cockroach-temp973375792
external I/O path:   /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster589129669/3/extern
store[0]:            path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster589129669/3
status:              restarted pre-existing node
clusterID:           91c48483-7eac-4918-84e1-8a069a3f3930
nodeID:              4
CockroachDB node starting at 2018-07-23 05:57:25.923620497 +0000 UTC (took 0.7s)
build:               CCL v2.1.0-alpha.20180702-734-g54a811d @ 2018/07/23 05:45:13 (go1.10.3)
admin:               http://127.0.0.1:26258
sql:                 postgresql://root@127.0.0.1:26257?sslmode=disable
logs:                /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/TestClusterRecovery/runMode=local/1
temp dir:            /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster589129669/1/cockroach-temp584756995
external I/O path:   /home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster589129669/1/extern
store[0]:            path=/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/acceptance/.localcluster589129669/1
status:              restarted pre-existing node
clusterID:           91c48483-7eac-4918-84e1-8a069a3f3930
nodeID:              1
test logs left over in: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/artifacts/acceptance/logTestClusterRecovery561334910

```

Please assign, take a look and update the issue accordingly.

--- a/pkg/sql/create_test.go
+++ b/pkg/sql/create_test.go
@@ -194,7 +194,10 @@ func createTestTable(
 
 	for {
 		if _, err := db.Exec(tableSQL); err != nil {
-			if testutils.IsSQLRetryableError(err) {
+			// Scenario where an ambiguous commit error happens is described in more
+			// detail in
+			// https://reviewable.io/reviews/cockroachdb/cockroach/10251#-KVGGLbjhbPdlR6EFlfL
+			if testutils.IsError(err, "result is ambiguous") {
 				continue
 			}
 			t.Errorf("table %d: could not be created: %s", id, err)

--- a/pkg/sql/distsqlrun/flow_registry_test.go
+++ b/pkg/sql/distsqlrun/flow_registry_test.go
@@ -573,3 +573,33 @@ func TestSyncFlowAfterDrain(t *testing.T) {
 	}
 	flow.Cleanup(ctx)
 }
+
+// TestInboundStreamTimeoutIsRetryable verifies that a failure from an inbound
+// stream to connect in a timeout is considered retryable by
+// testutils.IsSQLRetryableError.
+// TODO(asubiotto): This error should also be considered retryable by clients.
+func TestInboundStreamTimeoutIsRetryable(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+
+	fr := makeFlowRegistry(0)
+	wg := sync.WaitGroup{}
+	rb := &RowBuffer{}
+	inboundStreams := map[StreamID]*inboundStreamInfo{
+		0: {
+			receiver:  rb,
+			waitGroup: &wg,
+		},
+	}
+	wg.Add(1)
+	if err := fr.RegisterFlow(
+		context.Background(), FlowID{}, &Flow{}, inboundStreams, 0, /* timeout */
+	); err != nil {
+		t.Fatal(err)
+	}
+	wg.Wait()
+	if _, meta := rb.Next(); meta == nil {
+		t.Fatal("expected error but got no meta")
+	} else if !testutils.IsSQLRetryableError(meta.Err) {
+		t.Fatalf("unexpected error: %v", meta.Err)
+	}
+}

--- a/pkg/testutils/error.go
+++ b/pkg/testutils/error.go
@@ -71,7 +71,7 @@ func IsPError(pErr *roachpb.Error, re string) bool {
 func IsSQLRetryableError(err error) bool {
 	// Don't forget to update the corresponding test when making adjustments
 	// here.
-	return IsError(err, "(connection reset by peer|connection refused|failed to send RPC|rpc error: code = Unavailable|EOF|result is ambiguous)")
+	return IsError(err, "(no inbound stream connection|connection reset by peer|connection refused|failed to send RPC|rpc error: code = Unavailable|EOF|result is ambiguous)")
 }
 
 // Caller returns filename and line number info for the specified stack

